### PR TITLE
New: pinning pages

### PR DESCRIPTION
**Pull request by AndriiPopov3**

### CODE REVIEW

File: backend/src/api/routes/page-route.ts [hunk 1]
```diff
@@ -216,4 +224,14 @@ router.get(
   run((req) => getTableOfContentsShared(req.query.token)),
 );
 
+router.post(
+  '/pin/:id',
```
Comment (AlikRakhmonov) [inline backend/src/api/routes/page-route.ts]:
```suggestion
  '/:id/pin',
```

File: backend/src/api/routes/page-route.ts [hunk 2]
```diff
@@ -216,4 +224,14 @@ router.get(
...
+  run((req) => pinPage(req.userId, req.params.id)),
+);
+
+router.post(
+  '/unpin/:id',
```
Comment (AlikRakhmonov) [inline backend/src/api/routes/page-route.ts]:
```suggestion
  '/:id/unpin',
```

**Review by AlikRakhmonov — CHANGES_REQUESTED** [review]

**Review by AlikRakhmonov — APPROVED** [review]

File: backend/src/common/mappers/page/map-page-to-ipage.ts [hunk 3]
```diff
@@ -28,6 +29,15 @@ export const mapPageToIPage = (page: Page): IPage => {
...
 
+  const mappedPinnedUsers = pinnedUsers?.map((user) => {
+    return {
+      id: user.id,
+      fullName: user.fullName,
+      email: user.email,
+      avatar: user.avatar,
+    };
+  });
+
```
Comment (ovchynnikov) [inline backend/src/common/mappers/page/map-page-to-ipage.ts]: Please try this example locally if it works 

```suggestion
  const mappedPinnedUsers = pinnedUsers?.map(({ id, fullName, email, avatar }) => {
    return {
      id,
      fullName,
      email,
      avatar,
    };
  });

```

File: backend/src/services/page.service.ts [hunk 4]
```diff
@@ -238,6 +239,20 @@ export const getPages = async (
...
+  const userRepository = getCustomRepository(UserRepository);
+  const { pinnedPages } = await userRepository.findById(userId);
+  const filteredPages = pinnedPages.filter(
+    (page) => page.workspaceId === workspaceId,
+  );
+  const pagesToShow = mapPagesToPagesNavWithoutChildren(filteredPages);
+
+  return pagesToShow;
+};
+
```
Comment (ovchynnikov) [inline backend/src/services/page.service.ts]:
```suggestion
export const getPinnedPages = async (
  userId: string,
  workspaceId: string,
): Promise<IPageNav[]> => {
  const { pinnedPages } = await getCustomRepository(UserRepository).findById(userId);
  const filteredPages = pinnedPages.filter(
    (page) => page.workspaceId === workspaceId,
  );
  const pagesToShow = mapPagesToPagesNavWithoutChildren(filteredPages);

  return pagesToShow;
};

```

**Review by ovchynnikov — APPROVED** [review]

**Pull request merged**